Carolina Linux workflow fails when testing for the first time

When building the wheels, the `Test Carolina` step fails due to permission denied.
Re-running the failing tests always succeeds.

Ref.
https://github.com/equinor/Carolina/actions/runs/7169163880/job/19670223507

```
        # Write the configuration file
>       with open(infile, 'w') as out:
E       PermissionError: [Errno 13] Permission denied: 'dakota.in'

myvenv/lib/python3.9/site-packages/dakota.py:147: PermissionError
----------------------------- Captured stdout call -----------------------------

### Check run into exception.
=============================== warnings summary ===============================
carolina-main/tests/test_dakota.py:30
  /home/runner/work/Carolina/Carolina/carolina-main/tests/test_dakota.py:30: PytestCollectionWarning: cannot collect test class 'TestDriver' because it has a __init__ constructor (from: tests/test_dakota.py)
    class TestDriver(DakotaBase):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED carolina-main/tests/test_dakota.py::TestCase::test_dakota - PermissionError: [Errno 13] Permission denied: 'dakota.in'
FAILED carolina-main/tests/test_dakota.py::TestCase::test_dakota_exception - PermissionError: [Errno 13] Permission denied: 'dakota.in'
```

--- a/.github/workflows/bundle_with_dakota_linux.yml
+++ b/.github/workflows/bundle_with_dakota_linux.yml
@@ -103,7 +103,9 @@ jobs:
       run: |
         source myvenv/bin/activate
         pip install pytest numpy
-        python -m pytest ./carolina-main/tests
+        cd carolina-main/tests
+        python -m pytest .
+        cd -
 
     - name: Upload wheel as artifact
       if: always()

--- a/.github/workflows/bundle_with_dakota_macos.yml
+++ b/.github/workflows/bundle_with_dakota_macos.yml
@@ -242,6 +242,7 @@ jobs:
     - uses: actions/upload-artifact@v3
       id: cache-carolina_wheel
       with:
+        name: macOS_python_wheels
         path: /tmp/carolina_dist/carolina*
 
   tests:

